carina-core/src/main/java/com/qaprosoft/carina/core/foundation/AbstractTest.java

```diff
@@ -742,11 +731,13 @@ private void generateMetadata() {
                 PrintWriter out = null;
                 try {
                     out = new PrintWriter(file);
+                    out.append(JsonUtils.toJson(MetadataCollector.getAllCollectedData().get(key)));
+                    out.flush();
                 } catch (FileNotFoundException e) {
                     LOGGER.error("Unable to write metadata to json file: " + file.getAbsolutePath(), e);
+                } finally {
+                    out.close();
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'out' is nullable here [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://demo.qaprosoft.com/sonarqube/coding_rules#rule_key=squid%3AS2259)